### PR TITLE
Add Firestore listener to Candidate Decider page 

### PR DESCRIPTION
**Pull request by henry-li-06**
### Summary 

Add Firestore listener to listen to the Candidate Decider document in our Firestore database. This listener should listen to real-time changes and should update the information on the page in real-time. 

### Test Plan 
Visit the deploy preview -- Try adding/updating a comment or a review. The comment or review should update on the side panel after a brief period. This should indicate that the listener is working since the frontend no longer directly updates the state anymore.

### CODE REVIEW

File: common-types/index.d.ts [hunk 1]
```diff
@@ -145,3 +145,30 @@ interface DevPortfolioSubmission {
...
+
+interface DBCandidateDeciderInstance {
+  readonly name: string;
+  readonly headers: string[];
+  readonly candidates: DBCandidateDeciderCandidate[];
+  readonly uuid: string;
+  readonly authorizedMembers: firestore.DocumentReference[];
+  readonly authorizedRoles: Role[];
+  isOpen: boolean;
+}
```
Comment (henry-li-06) [inline common-types/index.d.ts:174]: These were moved here from `backend/src/DataTypes.d.ts` since these types are referenced in the frontend now

**Review by airaagrawal1 — COMMENTED** [review]
I tried adding 2 comments and my most recent one overwrote my older comment. Just wanted to double check that was intended functionality. Also, the comment box did not clear after pressing 'save'.

Screen Shot 2022-04-17 at 12 44 28 PM

Screen Shot 2022-04-17 at 12 44 58 PM

**Comment by henry-li-06** [comment]
> I tried adding 2 comments and my most recent one overwrote my older comment. Just wanted to double check that was intended functionality. Also, the comment box did not clear after pressing 'save'.

Yeah that should be the intended functionality. I believe that's how the original Candidate Decider works

**Review by airaagrawal1 — APPROVED** [review]
LGTM!

**Pull request merged**